### DISCUSSION TITLE
Allow Supporters to customize their chat name colour

### DISCUSSION THREAD

**Discussion by Vortex100K** [ppy/osu#19757] [Ideas]
![osu_2022-08-14_12-05-23](https://user-images.githubusercontent.com/79509710/184533882-00583592-d57f-4fbd-b507-9960825be7b1.png)

I think making this a feature would be pretty cool 

Non-supporters still have randomly generated colours 
supporters have the ability to customize the colour with a colour wheel or by inputting a hex

it would be an extra feature for supporters and drawing a little bit more players to buy supporter at the same time

I would love to make my name like white or #f7024c that would look so cool lol

**Comment by nagi-desuuu** [comment]
Yeah, but there still needs to be a way to determine if a user has supporter at a glance.
Sure, custom colors is great, but the colors in stable reflected if a user has supporter at a glance. If we're having custom colors, we still need to find some way to instantly tell if they have it or not.

**Reply by dragonGod97677** [reply]
> Yeah, but there still needs to be a way to determine if a user has supporter at a glance. Sure, custom colors is great, but the colors in stable reflected if a user has supporter at a glance. If we're having custom colors, we still need to find some way to instantly tell if they have it or not.

Why not adding a star badge next to user name

**Reply by nagi-desuuu** [reply]
Hold on, don't supporters have that filled-in box inside their name?

**Reply by Vortex100K** [reply]
> Hold on, don't supporters have that filled-in box inside their name?

I don't think so, I'm pretty sure only people with profile tags like BN , Staff etc get those filled in boxes 
I could be wrong though I've gone on lazer chat only like 4 times lol

**Reply by Vortex100K** [reply]
> > Yeah, but there still needs to be a way to determine if a user has supporter at a glance. Sure, custom colors is great, but the colors in stable reflected if a user has supporter at a glance. If we're having custom colors, we still need to find some way to instantly tell if they have it or not.
> 
> 
> 
> Why not adding a star badge next to user name

Thats sounds nice a little star badge or the osu supporter icon!

**Reply by cdwcgt** [reply]
It looks like the setting of the custom color requires a modification of osu-web, but the star badge can be implemented directly? If there is a good design

**Reply by Remyria** [reply]
imo, it should be like twitch, a palette to pick from as a free user and full RGB for supporter (idk about what direction to take to deal with hard to read colours)

every time I open lazer and I remember I can't pick my colour, I'm sad

**Reply by SirHonkkk** [reply]
just put a little heart next to their name

**Comment by Corne2Plum3** [comment]
I thought about something to both allow everyone to change their chat color and see which is supporter or not.

The idea is giving to supporter GRADIENT color. Supporters can choose a second color for their names, and these colors will be displayed as a gradient, something like this:

![chat_concept](https://user-images.githubusercontent.com/57914482/202911956-583db404-e177-4735-93d3-50249e6d71fc.jpg)

The advantage with it is it becomes easier to see who is supporter or not, and allow everyone to set his color.

**Reply by ghost** [reply]
would you be able to choose any color, or just pick from a palette?

**Reply by Corne2Plum3** [reply]
Depends on how the color name are implemented. It's just there would be 2 colors to pick instead of one.

But imo if custom colors are going to be implemented, I think it's going to be a palette, to avoid people choosing unreadable and confusing color, such as picking the same color than the background.

**Reply by Remyria** [reply]
that looks like what speedrun.com does

**Reply by I-AmBen** [reply]
I like this idea

**Comment by chromb** [comment]
Gonna revive this thread because with the addition of custom profile page colours, those can probably be shared with chat name colors (and i'm pretty sure that's a relatively easy addition)

**Reply by Remyria** [reply]
alternatively, you could copy and paste most of the code, slap it in the settings and have it only affect name colour

**Comment by DubstepBilly** [comment]
Bumping the thread cause I think this would be an awesome addition!
